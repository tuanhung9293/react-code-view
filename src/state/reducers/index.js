import {combineReducers} from 'redux'
import collections from './collections'
import medias from './medias'
import forms from './forms'
import ui from './ui'

const root = combineReducers({
  collections,
  medias,
  forms,
  ui
})

function rootReducer(state, action) {
  return root(state, action)
}

export default rootReducer