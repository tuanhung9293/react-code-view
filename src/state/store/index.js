import {createStore, applyMiddleware, compose} from 'redux'
import thunk from 'redux-thunk'
import {persistStore, persistReducer} from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import trackWindowMiddleware from 'state/middleware/trackWindowMiddleware'
import rootReducer from '../reducers'
import {startTrackingWindow} from '../actions/uiAction'

const middlewares = [
  thunk,
  trackWindowMiddleware
]

const persistConfig = {
  key: 'root',
  storage: storage,
  whitelist: ['collections']
}

const persistedReducer = persistReducer(persistConfig, rootReducer)

const isDev = process.env.NODE_ENV === 'development'

const composeEnhancers =
  (isDev && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose

export function createBaseStore(initialState = {}) {
  return createStore(
    persistedReducer,
    {
      ...initialState
    },
    composeEnhancers(applyMiddleware(...middlewares))
  )
}

const store = createBaseStore()
store.dispatch(startTrackingWindow())

const persistor = persistStore(store)

export {store, persistor}
