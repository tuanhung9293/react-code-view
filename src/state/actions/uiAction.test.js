import {
  startTrackingWindow,
  stopTrackingWindow,
  updateWindowSize
} from './uiAction'
import {createBaseStore} from 'state/store'

it('creates an action to startTrackingWindow', () => {
  expect(startTrackingWindow()).toMatchSnapshot()
})

it('creates an action to stopTrackingWindow', () => {
  expect(stopTrackingWindow()).toMatchSnapshot()
})

describe('integration', () => {
  it('updateWindowSize', () => {
    const store = createBaseStore({
      ui: {
        windowSize: {
          width: 666,
          height: 666
        }
      }
    })
    store.dispatch(updateWindowSize({width: 778, height: 877}))
    expect(store.getState().ui).toMatchSnapshot()
  })
})