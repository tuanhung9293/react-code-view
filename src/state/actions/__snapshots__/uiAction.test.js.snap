// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates an action to startTrackingWindow 1`] = `
Object {
  "type": "START_TRACKING_WINDOW",
}
`;

exports[`creates an action to stopTrackingWindow 1`] = `
Object {
  "type": "STOP_TRACKING_WINDOW",
}
`;

exports[`integration updateWindowSize 1`] = `
Object {
  "windowSize": Object {
    "height": 877,
    "width": 778,
  },
}
`;
